temfpy/optimization.py

```diff
@@ -1,20 +1,113 @@
+"""Test capabilities for optimization.
+
+This module contains a host of models and functions often used for testing optimization algorithms.
+
+"""
 import numpy as np
 import math
 
 
 def ackley(x, a=20, b=0.2, c=2 * math.pi):
     r"""Ackley function.
+
+    This function was proposed by David Ackley in [A1987]_.
+    Originally, it was formulated only for the two-dimensional case;
+    it is characterized by an almost flat outer region and a central hole or peak
+    where modulations become more and more influential. The function has
+    its global minimum :math:`f(x) = 0` at :math:`x = (0, \dots, 0)`.
+
+    Parameters
+    ----------
+
+    x : array_like
+        Input domain with dimension d. It is usually evaluated on the hypercube
+        :math:`x_i\in [-32.768, 32.768]`, for all :math:`i = 1, \dots, d`.
+
+    float a, b, c
+        Recommended variable values are 20, 0.2, 2π, respectively.

```

Comment:
This fix is ugly, but it's the only way I can get it to work the way you want. 

Also, default value should come after the parameter description, as seen in [numpydoc docstring guide](https://numpydoc.readthedocs.io/en/latest/format.html#sections).